File: Udemy - Machine Learning A-Z/Part 1 - Data Preprocessing/data_preprocessing_template.R
# Data Preprocessing Template

# Importing the dataset
dataset = read.csv('C:/Users/a0687514/Downloads/Desktop/Data Science/Machine Learning A-Z - Udemy/Part 1 - Data Preprocessing/Data.csv')

# Taking care of missing data
dataset$Age = ifelse(is.na(dataset$Age),
                     ave(dataset$Age, FUN = function(x) mean(x, na.rm = TRUE)),
                     dataset$Age)
dataset$Age

dataset$Salary = ifelse(is.na(dataset$Salary),
                        ave(dataset$Salary, FUN = function(x) mean(x, na.rm = TRUE)),
                        dataset$Salary)
dataset$Salary

# Encoding categorical data
dataset$Country = factor(dataset$Country,
                         levels = c('France', 'Spain', 'Germany'),
                         labels = c(1, 2, 3))
dataset$Country
dataset$Purchased = factor(dataset$Purchased,
                           levels = c('No', 'Yes'),
                           labels = c(0, 1))
dataset$Purchased

# Splitting the dataset into the Training set and Test set
# install.packages('caTools')
library(caTools)
"set.seed sets is used to remember the results whenever random sampling is used"
set.seed(123)
"'SplitRatio' is the % of data in training set"
split = sample.split(dataset$Purchased, SplitRatio = 0.8)
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)

# Feature Scaling
training_set[,2:3] = scale(training_set[,2:3])
test_set[,2:3] = scale(test_set[,2:3])

File: Udemy - Machine Learning A-Z/Part 1 - Data Preprocessing/data_preprocessing_template.py
# Data Preprocessing Template

# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing the dataset
dataset = pd.read_csv('Data.csv')
X = dataset.iloc[:, :-1].values
y = dataset.iloc[:, 3].values

# Taking care of missing data
from sklearn.preprocessing import Imputer
"""Select 'Imputer' and Hit Ctrl+I to see all input parameters"""
imputer = Imputer(missing_values = 'NaN', strategy = 'mean', axis = 0)
imputer = imputer.fit(X[:, 1:3])
X[:, 1:3] = imputer.transform(X[:, 1:3])
X

# Encoding categorical data
# Encoding the Independent Variable
from sklearn.preprocessing import LabelEncoder, OneHotEncoder
labelencoder_X = LabelEncoder()
X[:, 0] = labelencoder_X.fit_transform(X[:, 0])
X
"""Select 'OneHotEncoder' and Hit Ctrl+I to see all input parameters"""
onehotencoder = OneHotEncoder(categorical_features = [0])
X = onehotencoder.fit_transform(X).toarray()
X
# Encoding the Dependent Variable
labelencoder_y = LabelEncoder()
y = labelencoder_y.fit_transform(y)
y

# Splitting the dataset into the Training set and Test set
from sklearn.cross_validation import train_test_split
"""'test_size' corresponds to % of data in test set"""
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2, random_state = 0)

# Feature Scaling
from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.transform(X_test)
"""It is important to fit the object on X_train first 
so that X_train and X_test are scaled on same basis"""
sc_y = StandardScaler()
y_train = sc_y.fit_transform(y_train)
